metric_learn/base_metric.py

```diff
@@ -296,6 +296,7 @@ def get_mahalanobis_matrix(self):
 
 class _PairsClassifierMixin(BaseMetricLearner):
 
+  classes_ = [-1, 1]
```

Comment:
We need estimators to have a `classes_` attribute for `CalibratedClassifierCV` to work. Typically, scikit-learn estimators deduce it at fit time, but in our case we will always have these ones (see also issue #164). That's why I put it as a class attribute, to be more explicit.

Classes have to be in this order (`[-1, 1]`)(i.e. sorted), otherwise there is a bug: because here https://github.com/scikit-learn/scikit-learn/blob/1a850eb5b601f3bf0f88a43090f83c51b3d8c593/sklearn/calibration.py#L312-L313:
`self.label_encoder_.classes_` is a sorted list of classes (so it'll be `[-1, 1]` in every cases). So when it transforms `self.base_estimator.classes_` it will return `[0, 1]` if our estimator has `[-1, 1]` as classes, and `[1, 0]` if it has `[1, -1]`. In the latter case, it will return `IndexError` because here (https://github.com/scikit-learn/scikit-learn/blob/1a850eb5b601f3bf0f88a43090f83c51b3d8c593/sklearn/calibration.py#L357) it will try to reach the column n°1 (and not 0) of `Y` (which does not exist).

Also, we have to put -1 and 1 in this order so that for `label_binarizer` (which is called by `CalibratedClassifierCV`), -1 (first element of `[-1, 1]`) will be considered as the negative label, and 1 will be considered as the positive label. (That's how `label_binarizer` work, see the example below)

Example: (warning: pos_label and neg_label are not to say what input classes are pos or neg, as one could think, but rather how we want them to appear in the output)
```python
In [1]: from sklearn.preprocessing import label_binarize 
   ...: print(label_binarize([-1, 1, -1], [-1, 1], pos_label=2019, neg_label=2018))                                                                                                                                                                                                        
[[2018]
 [2019]
 [2018]]
```
And:

```python
In [1]: from sklearn.preprocessing import label_binarize 
   ...: print(label_binarize([-1, 1, -1], [1, -1], pos_label=2019, neg_label=2018))                                                                                                                                               
[[2019]
 [2018]
 [2019]]
```